File: examples/java-calculator-junit5/src/test/java/io/cucumber/examples/junit5/calculator/DateStepDefinitions.java
package io.cucumber.examples.junit5.calculator;

import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

import java.util.Date;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class DateStepDefinitions {

    private String result;
    private DateCalculator calculator;

    @Given("^today is ([0-9]{4}-[0-9]{2}-[0-9]{2})$")
    public void today_is(Date date) {
        calculator = new DateCalculator(date);
    }

    @When("^I ask if ([0-9]{4}-[0-9]{2}-[0-9]{2}) is in the past$")
    public void I_ask_if_date_is_in_the_past(Date date) {
        result = calculator.isDateInThePast(date);
    }

    @Then("^the result should be (yes|no)$")
    public void the_result_should_be(String expectedResult) {
        assertEquals(expectedResult, result);
    }

}

File: examples/calculator-kotlin-junit5/src/test/kotlin/io/cucumber/examples/calculator/DateStepDefinitions.kt
package io.cucumber.examples.calculator

import io.cucumber.examples.calaculator.DateCalculator
import io.cucumber.java.en.Given
import io.cucumber.java.en.Then
import io.cucumber.java.en.When
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.util.Locale
import kotlin.test.assertEquals

class DateStepDefinitions {

    private lateinit var  result: String

    private lateinit var calculator: DateCalculator


    @Given("today is {}")
    fun todayIs(date: String) {
        val date1 = LocalDate.parse(date)
        calculator = DateCalculator(now = date1)
    }

    @When("I ask if {} is in the past")
    fun iAskIfDateIsInPast(date: String) {
        val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("MMM d, yyyy", Locale.ENGLISH)
        val date2 = LocalDate.parse(date,formatter)
        result = calculator.isDateInThePast(date2)
    }

    @Then("the result should be {string}")
    fun theResultShouldBeYes(answer: String) {
        assertEquals(answer, result)
    }
}
